lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -174,25 +196,35 @@ defmodule Kernel.LexicalTracker do
 
   # Callbacks helpers
 
-  defp add_reference(references, module, :runtime) when is_atom(module) do
-    case :maps.find(module, references) do
-      {:ok, _} -> references
-      :error -> :maps.put(module, :runtime, references)
-    end
-   end
-
+  defp add_reference(references, module, :runtime) when is_atom(module),
+    do: map_put_new(references, module, :runtime)
   defp add_reference(references, module, :compile) when is_atom(module),
-    do: :maps.put(module, :compile, references)
+    do: map_put(references, module, :compile)
+
+  defp add_remote_dispatch(dispatches, module, fa, line, :runtime) when is_atom(module),
+    do: map_put_new(dispatches, {module, fa, line}, :runtime)
+  defp add_remote_dispatch(dispatches, module, fa, line, :compile) when is_atom(module),
+    do: map_put(dispatches, {module, fa, line}, :compile)
 
   # In the map we keep imports and aliases.
   # If the value is a line, it was imported/aliased and has a pending warning
   # If the value is true, it was imported/aliased and used
   defp add_directive(directives, module_or_mfa, line, warn, tag) do
     marker = if warn, do: line, else: true
-    :maps.put({tag, module_or_mfa}, marker, directives)
+    map_put(directives, {tag, module_or_mfa}, marker)
   end
 
   defp add_dispatch(directives, module_or_mfa, tag) do
-    :maps.put({tag, module_or_mfa}, true, directives)
+    map_put(directives, {tag, module_or_mfa}, true)
   end
+
+  defp map_put_new(map, key, value) do
+    case :maps.find(key, map) do
+      {:ok, _} -> map
+      :error -> map_put(map, key, value)
+    end
+  end
+
+  defp map_put(map, key, value),
```

Comment:
Honestly, I don't see any difference, it just brings a question why we have this function defined (and it simply more code for no gain). :bowtie: